config/agent.go

```diff
@@ -0,0 +1,76 @@
+package config
+
+import (
+	"flag"
+	"log"
+	"log/slog"
+	"os"
+	"time"
+
+	"github.com/caarlos0/env/v11"
+	"github.com/mihailtudos/metrickit/pkg/flags"
+)
+
+const DefaultReportInterval = 10
+const DefaultPoolInterval = 2
+
+var serverAddr *flags.ServerAddr
```

Comment:
You don't really need any env vars, just use the structure fields

Reply:
I connected it to the pipeline, I am getting a duplication err but that's because I have a similar implementation for working with Gauge metrics and Counter metrics but other than that It seems ok.